rubin_scheduler/scheduler/detailers/dither_detailer.py

```diff
@@ -51,24 +52,42 @@ def __init__(self, max_dither=0.7, seed=42, per_night=True, nnights=7305):
         self.max_dither = np.radians(max_dither)
         self.per_night = per_night
         self.rng = np.random.default_rng(seed)
+
+        self.n_positions = copy.copy(nnights)
```

Comment:
Ok, this is for my own learning -- why do you need a copy here? 
Isn't `nnights` just an integer, so assigning self.n_positions to the integer would always be a copy? 
I'm assuming that the copy was needed because of the if not per_night clause below, but don't see why unless I'm not understanding what data type "nnights" is.